runtime/protocol/http-client/common/src/aws/smithy/kotlin/runtime/http/interceptors/requestcompression/CompressionAlgorithm.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.http.interceptors.requestcompression
+
+import aws.smithy.kotlin.runtime.http.request.HttpRequest
+
+/**
+ * Represents a compression algorithm. Is used by algorithms compressing request payloads on operations with the
```

Comment:
nit: `CompressionAlgorithm` "Is used by algorithms..." is odd, I would say these algorithms are "used by an interceptor..."